toolset/setup/linux/systools/composer.sh

```diff
@@ -4,12 +4,16 @@ RETCODE=$(fw_exists ${IROOT}/php-composer.installed)
 [ ! "$RETCODE" == 0 ] || { return 0; }
 
 fw_depends php
-fw_get https://getcomposer.org/installer -O composer-installer.php
 
 mkdir -p php-composer
+cd php-composer
+
+fw_get https://getcomposer.org/installer -O composer-installer.php
 
-# Use the PHP from our IROOT directory
-PHP_HOME=${IROOT}/php-5.5.17
-${PHP_HOME}/bin/php composer-installer.php --install-dir=$IROOT
+# Use the PHP and composer from our PHP_HOME directory and 
+# COMPOSER_HOME directories (should be specified in frameworks 
```

Comment:
Instead of saying "should be specified in ...", either a) throw a big warning that this variable is required to depend on PHP, or b) just provide a default on the variable using something like this. That way the requirement is baked into the codebase instead of just as a comment

```
PHP_HOME=${PHP_HOME:-${IROOT}/php-5.5.17}
```

See [here](http://stackoverflow.com/questions/2013547/assigning-default-values-to-shell-variables-with-a-single-command-in-bash) for more info